CMakeLists.txt

```diff
@@ -290,6 +290,15 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
   target_compile_options(ds2 PRIVATE /MP)
 endif ()
 
+if (COVERAGE AND (CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang"))
+  target_compile_options(ds2 PRIVATE -fprofile-arcs -ftest-coverage)
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
+
+  if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+    target_link_libraries(ds2 gcov)
```

Comment:
It compiles and links locally, but not on travis (can't find libprofiler on travis).